src/components/Certificate.tsx

```diff
@@ -34,15 +36,11 @@ export const CertificateComponent = ({
     userName,
   });
 
-  // const handleDownloadPDF = async () => {
-  //   const downloadUrl = certificatePdfUrl;
-  //   const a = document.createElement('a');
-  //   a.href = downloadUrl!;
-  //   a.download = 'certificate.pdf';
-  //   document.body.appendChild(a);
-  //   a.click();
-  //   document.body.removeChild(a);
-  // };
+  useEffect(() => {
+    if (certificateImageUrl) {
+      setLoading(false);
+    }
+  }, [certificateImageUrl]);
 
   const handleDownloadPNG = async () => {
     const downloadUrl = certificateImageUrl;
```

Comment:
we can also here add setLoading(true)

and at the end of the function set loading to false?

this way if the user clicks download multiple times the image doesnt get dowaloaded multiple times right ? 

do it for all the three functions

Reply:
but then the buttons for share on linkedin and twitter would be disabled as well right? did you mean for me to make three states for three functions seperately?

Reply:
@siinghd should i put a 1 second timeout after the click to fix this instead? or go with the 3 separate states? or did you mean some other way?

Reply:
> @siinghd should i put a 1 second timeout after the click to fix this instead? or go with the 3 separate states? or did you mean some other way?

You already have loading state

Reply:
> You already have loading state

@hkirat yes but that will disable all three of the buttons after clicking on download

Reply:
@siinghd